pallets/compliance-manager/src/lib.rs

```diff
@@ -957,10 +855,8 @@ impl<T: Trait> ComplianceManagerTrait<T::Balance> for Module<T> {
         // gets checked only once and if it is valid then its result is tied up with the conditions result.
         //
         // Note - Due to this check `ConditionType::HasValidProofOfInvestor` is an implicit transfer condition and it only
-        // lookup for the claims those are provided by the trusted_claim_issuers.
-        if Self::implicit_requirements_status(ticker) == ImplicitRequirementStatus::Active
-            && !Self::is_sender_and_receiver_has_valid_scope_claim(ticker, from_did_opt, to_did_opt)
-        {
+        // lookup for the claims those are provided by the user itself.
+        if !Self::is_sender_and_receiver_has_valid_scope_claim(ticker, from_did_opt, to_did_opt) {
```

Comment:
This verifies the ZK proof. We don't need to do that anymore since ZK proofs are now verified when we add them. We don't need to give any special treatment to these claims anymore, we can use the generic `ConditionType::IsPresent` for cleaner code.

Reply:
This doesn't verifies the ZK proof anymore, It only looks whether the claim exists or not. We can't even use the `ConditionType::IsPresent` as we don't have the explicit claim here even if we want to create a claim we can't do it as we don't know the scopeId of the user.

Reply:
This calls `has_scope_claim`>`is_condition_satisfied`>`HasValidProofOfInvestor::run`>`has_valid_proof_of_investor`>`ValidProofOfInvestorProposition::evaluate_claim`>`verify_proof`.

Am I missing something here?

Reply:
It was the old process now it changes to 

`has_scope_claim>is_condition_satisfied>HasValidProofOfInvestor::run>has_scope_claim_exists`

Reply:
In that case, I wonder if this needs to be a condition type at all. It will be much simpler to just add a different function for it since it's something the system checks, not something the user defines. There's just too much misdirection right now. Anyway, we can tackle it in a different PR. It looks good from this PR's context. I had old code checked out earlier.